### PR TITLE
Un-revert config rename changes.

### PR DESCRIPTION
**Pull request by squeed**
This reverts commit 62d475bd365ae7e8d2b527b6b95c6a2a44262d06, reversing
changes made to 22c3f02425be355ea74641cd36fade6d478532d4.

This brings back the config object rename in #103.

### CODE REVIEW

**Comment by squeed** [comment]
Need https://github.com/openshift/release/pull/2922 to merge before CI will pass.

**Comment by crawford** [comment]
Rate-limiting in the last [failure](https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_cluster-network-operator/105/pull-ci-openshift-cluster-network-operator-master-e2e-aws/794):

```
level=fatal msg="failed to fetch Terraform Variables: failed to fetch dependency of \"Terraform Variables\": failed to fetch dependency of \"Bootstrap Ignition Config\": failed to fetch dependency of \"Common Manifests\": failed to generate asset \"DNS Config\": getting public zone for \"origin-ci-int-aws.dev.rhcloud.com\": listing hosted zones: Throttling: Rate exceeded\n\tstatus code: 400, request id: b7b21bcb-362a-11e9-8be7-a7e996f87a18"
```

We'll try again later.

/joke

**Comment by squeed** [comment]
nat gateway creation timeout.
/retest

File: pkg/network/cluster_config.go [hunk 1]
```diff
@@ -97,8 +98,8 @@ func StatusFromOperatorConfig(operConf *netopv1.NetworkConfigSpec) configv1.Netw
 	}
 
 	// Determine the MTU from the provider
-	switch operConf.DefaultNetwork.Type {
-	case netopv1.NetworkTypeOpenShiftSDN:
+	switch strings.ToLower(string(operConf.DefaultNetwork.Type)) {
+	case strings.ToLower(string(netopv1.NetworkTypeOpenShiftSDN)):
```
Comment (danwinship) [inline pkg/network/cluster_config.go]: bleah. can't we canonicalize on read so that the rest of the code only has to deal with the correct form?

File: pkg/network/openshift_sdn.go [hunk 2]
```diff
@@ -143,7 +143,7 @@ func fillOpenShiftSDNDefaults(conf, previous *netv1.NetworkConfigSpec, hostMTU i
 	// However, this can never change, so we always prefer previous.
 	if sc.MTU == nil {
 		var mtu uint32 = uint32(hostMTU) - 50 // 50 byte VXLAN header
-		if previous != nil {
+		if previous != nil && previous.DefaultNetwork.OpenShiftSDNConfig != nil && previous.DefaultNetwork.OpenShiftSDNConfig.MTU != nil {
```
Comment (danwinship) [inline pkg/network/openshift_sdn.go]: this is an unrelated fix that should get its own commit with its own commit message

File: pkg/util/clusteroperator/status_manager.go [hunk 3]
```diff
@@ -85,8 +85,8 @@ func (status *StatusManager) SetConfigSuccess() error {
 	status.configFailure = false
 	return status.Set(
 		&configv1.ClusterOperatorStatusCondition{
-			Type:    configv1.OperatorFailing,
-			Status:  configv1.ConditionFalse,
+			Type:   configv1.OperatorFailing,
+			Status: configv1.ConditionFalse,
```
Comment (danwinship) [inline pkg/util/clusteroperator/status_manager.go]: (this is also unrelated and yeah, we need a "verify" script/target)

File: pkg/network/openshift_sdn.go (hunk 2 above)
Comment (squeed) [reply pkg/network/openshift_sdn.go]: bah, fine.

File: pkg/util/clusteroperator/status_manager.go (hunk 3 above)
Comment (squeed) [reply pkg/util/clusteroperator/status_manager.go]: We totally have one, and I'm not sure why it's been passing.

Comment (squeed) [reply pkg/util/clusteroperator/status_manager.go]: Ah, because it always exits 0.

**Comment by squeed** [comment]
Thanks, Dan, for keeping me honest. Added a proper `Canonicalize` function.

I decided against inserting it in to `Validate` because I don't think `Validate` should mutate, even though it takes a pointer.

**Comment by danwinship** [comment]
/lgtm
(btw I am OK with keeping the MTU nil check in this PR, it just should be a separate commit within the PR)
